### PR TITLE
Inspired by dotnet show

### PR DESCRIPTION
**Pull request by IEvangelist**
- Reordered *.csproj*, this will be the default start up project if not otherwise specified in the *.suo* file.
- Added System.Text.Json - replaced Newtonsoft.Json
- Added unit tests and fixed places where tests needed updating

Might not need #385

### CODE REVIEW

**Comment by ardalis** [comment]
Are we able to link to a recording of that show?

**Comment by IEvangelist** [comment]
> Are we able to link to a recording of that show?

Yes sir, here it is : https://www.twitch.tv/videos/634482950

File: src/ApplicationCore/Extensions/StringExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Text.Json;
+
+namespace Microsoft.eShopWeb
+{
+    public static class StringExtensions
```
Comment (efleming18) [inline src/ApplicationCore/Extensions/StringExtensions.cs:5]: Could we name this `JsonExtensions` instead?

Good call on pulling this out in to an Extensions class. I was looking through to see if we could set these settings globally so we don't have to have an Extensions class but it doesn't look like that is scheduled until .NET 5.

File: tests/UnitTests/ApplicationCore/Extensions/FooBar.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+namespace Microsoft.eShopWeb.UnitTests.ApplicationCore.Extensions
+{
+    public class Foo : IEquatable<Foo>
```
Comment (efleming18) [inline tests/UnitTests/ApplicationCore/Extensions/FooBar.cs:9]: Could we:

-  Give these some more meaningful names? Something like `TestParent` and `TestChild`? Something to be more descriptive of what the actual classes are.
- Split the two classes out in to their own files?

File: tests/UnitTests/ApplicationCore/Extensions/StringExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+            var json = foo.ToJson();
+            var result = json.FromJson<Foo>();
+            Assert.Equal(foo, result);
+        }
+
+        [
+            Theory,
+            InlineData("{ \"id\": 9, \"name\": \"Another test\" }", 9, "Another test"),
+            InlineData("{ \"id\": 3124, \"name\": \"Ugh, this doesn't matter...\" }", 3124, "Ugh, this doesn't matter..."),
```
Comment (efleming18) [inline tests/UnitTests/ApplicationCore/Extensions/StringExtensions.cs:30]: Can you replace `"Ugh, this doesn't matter"` to something like `Another Tests 2`, or `Test Value 1`, something like that?

**Comment by efleming18** [comment]
Hey @IEvangelist - Do you want to make these changes or are you ok with it if I just make them since they are pretty minor?

**Comment by IEvangelist** [comment]
> Hey @IEvangelist - Do you want to make these changes or are you ok with it if I just make them since they are pretty minor?

Sorry, I totally forgot about these. I'm fine with you doing it. You could have made them "suggestions" then I could have just clicked a button to accept the changes, would have made it easier for both of us 😄 . If you don't get to it soon, I'll make it a TODO for this weekend. Thank you

**Pull request merged**